androidHyperskillApp/src/main/java/org/hyperskill/app/android/profile/view/dialog/AbstractBadgeDialogFragment.kt

```diff
@@ -0,0 +1,45 @@
+package org.hyperskill.app.android.profile.view.dialog
+
+import android.content.Context
+import androidx.annotation.ColorInt
+import androidx.core.content.ContextCompat
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import org.hyperskill.app.android.HyperskillApp
+import org.hyperskill.app.android.R
+import org.hyperskill.app.badges.domain.model.BadgeKind
+import org.hyperskill.app.badges.domain.model.BadgeRank
+import org.hyperskill.app.profile.view.BadgeImage
+import org.hyperskill.app.profile.view.BadgesViewStateMapper
+
+abstract class AbstractBadgeDialogFragment : BottomSheetDialogFragment() {
```

Comment:
Avoid inheritance. It limits you when some inheritor should have different behavior. It's better to replace abstract class with a util object. For example `BadgeDialogFormatter`.